### ISSUE TITLE
Scroll Reverser not functioning correctly after wake up from sleep

### ISSUE THREAD

**Issue by 16patsle** [pilotmoon/Scroll-Reverser#40]
First, thanks for this great app. It has helped me a lot I would say. Second, I have noticed that even after the recent update with the fix for #15, the detecting of trackpad as mouse after sleep, it will still occasionally detect the trackpad as a mouse. This happens much less frequently than before (now a couple times a week, with me traveling with my mac two times a day), but it does happen. My guess would be that the application for some reason is not restarting properly. Anyway, here are my system details, in case it is needed:

* MacBook Pro (Retina, 13-inch, Early 2015)
* MacOS Sierra 10.12.1
* Scroll Reverser 1.7.4

**Comment by invariant** [comment]
Ref: **[Conversation #5611](https://secure.helpscout.net/conversation/328538206/5611) / Anton**

**Comment by andrew-hill** [comment]
I'm seeing this happening even without sleep/wake events. While I'm using the computer, I'll randomly (about once or twice a day?) experience the scrolling direction switch for my Magic Trackpad.

* MacBook Pro (Retina, 15-inch, Late 2013)
* MacOS Sierra 10.12.4
* Scroll Reverser 1.7.4

I've set my system to natural scrolling, with Scroll Reverser configured to "Reverse Mouse" only, so I assume the Magic Trackpad is somehow getting re-detected as a mouse (though am not aware of any specific strings to search for in system log or way of getting this information from Scroll Reverser to check exactly why the scrolling direction flips).

**Comment by mikerenfro** [comment]
No Magic Trackpad here, just the built-in one on a MacBook Pro Retina. When I just now saw my trackpad reverse itself incorrectly, here's the first few lines from the debug log:

```
20:44:25.3 Log started [on yes][v yes][h no][trackpad yes][tablet yes][mouse/other yes]
20:44:25.3 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:25.5 Settings changed [on yes][v yes][h no][trackpad no][tablet yes][mouse/other yes]
20:44:41.2 0 [type gesture][touching 2]
20:44:41.3 1 [type gesture][touching 2]
```

I never went into the Preferences to change anything, but the first two lines clearly shows the trackpad setting flipping from yes to no on its own. Scroll Reverser 1.7.4 on Sierra 10.12.4.

**Comment by 16patsle** [comment]
@andrew-hill @mikerenfro I have the same problems as you. Originally it was only after wake up from sleep, but it has happened several times just randomly while browsing the web. No idea what's the cause, but the log you posted shows exactly the kind of behavior I experience.

**Comment by jrittenh** [comment]
+1 for trackpad reversing from "natural".  No amount of flipping the trackpad checkbox does anything.  Disabling and re-enabling doesn't help, either.  Quitting Scroll Reverser and launching it again does appear to resolve the issue.

**Comment by invariant** [comment]
Closing old issue. Please raise a new issue if any problems persist. Thanks!
